### ISSUE TITLE
Teams bot onMemberAdded messages without actual event

### ISSUE THREAD

**Issue by whyte624** [OfficeDev/Microsoft-Teams-Samples#997]
I have a Teams bot added to channel.
Bot is based on [Proactive Messaging Bot][1] from Microsoft GitHub.

There is a following peace of code: 

    this.onMembersAdded(async (context, next) => {
      const membersAdded = context.activity.membersAdded;
      
      for (let cnt = 0; cnt < membersAdded.length; cnt++) {
        if (membersAdded[cnt].id !== context.activity.recipient.id) {
          const welcomeMessage = `Welcome to the Proactive Bot sample.  Navigate to ${process.env.PROVISIONOUTPUT__BOTOUTPUT__SITEENDPOINT}/api/notify to proactively message everyone who has previously messaged this bot.`;
          await context.sendActivity(welcomeMessage);
        }
      }

      // By calling next() you ensure that the next BotHandler is run.
      await next();
    });

Basically, whenever new member is added, there will be a welcome message.

The problem is that sometimes we see in the channel welcome messages without any reason - no new member added.

What could be the reason?

  [1]: https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-proactive-messaging-teamsfx

**Comment by ChetanSharma-msft** [comment]
Hello @whyte624 - Thanks for raising your query.
We will look into it and let you know the updates.

**Comment by sayali-MSFT** [comment]
@Pawank-MSFT -Could you please help on this issue?

**Comment by whyte624** [comment]
Hi @sayali-MSFT!
I am not very experienced with teams bots. I just followed instructions ho it can be implemented and did it.
I do not think I have a network trace. How can I find it? Or maybe there are other ways to debug the problem?

```
        this.onMembersAdded(async (context, next) => {
            const membersAdded = context.activity.membersAdded;
            for (let cnt = 0; cnt < membersAdded.length; cnt++) {
                if (membersAdded[cnt].id !== context.activity.recipient.id) {
                    const welcomeMessage = 'Welcome!';
                    await context.sendActivity(welcomeMessage);
                }
            }

            // By calling next() you ensure that the next BotHandler is run.
            await next();
        });
```

Basically, sometimes without any changes in channel members we receive message 'Welcome!'.

For example few latest _messages out of the blue_ happened (GMT+2):
2023-12-12 16:35
2023-11-30 15:35
2023-11-23 17:35
2023-11-13 18:31
2023-11-7 14:57
2023-10-12 13:24

In fact this bot should not write anything in group channel only in private messages.
